scripts/zkai_d128_proof_artifact_backend_spike_gate.py

```
@@ -0,0 +1,882 @@
+
+
+def mutation_rows_for_tsv(payload: dict[str, Any]) -> list[dict[str, Any]]:
+    return [{column: case.get(column) for column in MUTATION_TSV_COLUMNS} for case in payload["cases"]]
+
+
+def _assert_repo_output_path(path: pathlib.Path) -> pathlib.Path:
+    resolved = path.resolve()
+    try:
+        resolved.relative_to(ROOT.resolve())
+    except ValueError as err:
+        raise D128BackendSpikeError(f"output path escapes repository: {path}") from err
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)